### PR TITLE
Dice loss branch

### PR DESCRIPTION
**Pull request by gamenerd457**
# Added dice loss 

Brief Description of the PR: 

dice loss can be used as a loss function for training segmentation models

Fixes # (issue)

## Type of change

- i have added the dice loss file to tensorflow_addons/losses
- i have added the test file for dice loss to tensorflow_addons/losses/tests

# How Has This Been Tested?
i have used unittest to test this code
*

### CODE REVIEW

**Comment by gamenerd457** [comment]
@googlebot i signed it

**Comment by bhack** [comment]
/cc @yarri-oss I think we already have a dice loss working also on 3d volumes /cc @fyangf :

https://github.com/tensorflow/models/blob/master/official/vision/beta/projects/volumetric_models/losses/segmentation_losses.py

So in the next release of:
`pip install tf-models-official`

We also had another dice loss at:
https://github.com/tensorflow/models/blob/master/research/object_detection/core/losses.py

**Comment by gamenerd457** [comment]
> /cc @yarri-oss I think we already have a dice loss working also on 3d volumes /cc @fyangf :
> 
> https://github.com/tensorflow/models/blob/master/official/vision/beta/projects/volumetric_models/losses/segmentation_losses.py
> 
> So in the next release of:
> `pip install tf-models-official`
> 
> We also had another dice loss at:
> https://github.com/tensorflow/models/blob/master/research/object_detection/core/losses.py

so should i close the pull request

**Comment by fyangf** [comment]
The dice loss under https://github.com/tensorflow/models/blob/master/official/vision/beta/projects/volumetric_models/losses/segmentation_losses.py is tailored for 3D volumetric model use cases. I'm ok with having a more generic implementation outside of that.

**Comment by bhack** [comment]
@gamenerd457 Please keep this open in the meantime as we would discuss this recurrent topic internally.
Another good place where this PR could be submitted is https://github.com/tensorflow/models/tree/master/official/vision/keras_cv/losses

**Comment by gamenerd457** [comment]
i opened this pr in https://github.com/tensorflow/models/tree/master/official/vision/keras_cv/losses but it is still  not reviewed

**Comment by bhack** [comment]
As https://github.com/tensorflow/models/pull/10249 was rejected cause:

> keras_cv is ramping down in this repo now. We do not accept changes to this folder. 

 we are talking with @qlzh727 and @fchollet about the roadmap.

**Comment by bhack** [comment]
@gamenerd457  If you are still interested in this please comment at https://github.com/keras-team/keras-cv/issues/296#issuecomment-1096952066

**Comment by bhack** [comment]
A PR now is at https://github.com/keras-team/keras-cv/pull/371 you can contribute there. Thanks
